packages/ember-toucan-core/src/components/form/controls/character-count.ts

```diff
@@ -0,0 +1,48 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+
+export interface ToucanFormControlCharacterCountComponentSignature {
+  Element: HTMLElement;
+  Args: {
+    /**
+     * The current number of characters inside the input
+     */
+    current?: number;
+
+    /**
+     * The max amount allowed for this input
+     */
+    max?: number;
+    id: string;
+  };
+}
+
+export default class ToucanFormControlCharacterCount extends Component<ToucanFormControlCharacterCountComponentSignature> {
+  constructor(
+    owner: unknown,
+    args: ToucanFormControlCharacterCountComponentSignature['Args']
+  ) {
+    assert(
+      'An "@id" argument is required, the id should be from the related input',
+      args.id
+    );
+    super(owner, args);
+  }
+
+  get initialCount() {
+    const id = this.args.id;
+    const input = <HTMLInputElement>document.getElementById(id);
```

Comment:
I would prefer to keep this component more "dumb". Can't we just make `current` a required argument? 

The normal flow of data is that we have state, and the DOM is a reflection of that state. Now we are basically pulling state (`input.value.length`) from the DOM. Might not be a huge issue here, but I would prefer to keep the other way around...

We could for example listen for `@onChange` of our `<Form::Controls::Input>`, set that as a `@tracked` property and have getter `get count()` that return the length of it. This we could then curry to the yielded character-count component, so it receives the count directly instead of assuming there must be an input somewhere and querying that.

Reply:
> We could for example listen for @onChange of our <Form::Controls::Input>, set that as a @tracked property and have getter get count() that return the length of it. This we could then curry to the yielded character-count component, so it receives the count directly instead of assuming there must be an input somewhere and querying that.

I like this idea!